google/cloud/pubsub/topic_admin_client.h

```diff
@@ -61,15 +61,24 @@ GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
  */
 class TopicAdminClient {
```

Comment:
We probably want to deprecate the class, instead of its member functions. e.g.:

https://github.com/googleapis/google-cloud-cpp/blob/f0af9a332320b5755ad5471cd15bd071edfa4264/google/cloud/spanner/database_admin_client.h#L120C7-L120C52

We should use `GOOGLE_CLOUD_CPP_DEPRECATED` instead of `[[deprecated]]`, to respect `GOOGLE_CLOUD_CPP_DISABLE_DEPRECATION_WARNINGS`:

https://github.com/googleapis/google-cloud-cpp/blob/f0af9a332320b5755ad5471cd15bd071edfa4264/google/cloud/internal/attributes.h#L37

Reply:
Ack, followed https://github.com/googleapis/google-cloud-cpp/pull/7359 to update deprecation stuff.